newIDE/app/src/Profile/CreateAccountDialog.js

```diff
@@ -76,119 +73,137 @@ export const getPasswordErrorText = (error: ?AuthError) => {
-                  ...this.state.form,
-                  username: value,
-                },
-              });
-            }}
-            allowEmpty
-          />
-          <TextField
-            value={this.state.form.email}
-            floatingLabelText={<Trans>Email</Trans>}
-            errorText={getEmailErrorText(error)}
-            fullWidth
-            required
-            onChange={(e, value) => {
-              this.setState({
-                form: {
-                  ...this.state.form,
-                  email: value,
-                },
-              });
-            }}
+  return (
+    <Dialog
+      title={<Trans>Create a new GDevelop account</Trans>}
+      actions={[
+        <FlatButton
+          label={<Trans>Back</Trans>}
+          disabled={createAccountInProgress}
+          key="close"
+          primary={false}
+          onClick={onClose}
+        />,
+        <LeftLoader
+          isLoading={createAccountInProgress || isValidatingUsername}
+          key="create-account"
+        >
+          <DialogPrimaryButton
+            label={<Trans>Create my account</Trans>}
+            primary
+            disabled={!canCreateAccount}
+            onClick={createAccount}
           />
-          <TextField
-            value={this.state.form.password}
-            floatingLabelText={<Trans>Password</Trans>}
-            errorText={getPasswordErrorText(error)}
-            type="password"
-            fullWidth
-            required
-            onChange={(e, value) => {
-              this.setState({
-                form: {
-                  ...this.state.form,
-                  password: value,
-                },
-              });
-            }}
+        </LeftLoader>,
+      ]}
+      secondaryActions={[
+        <FlatButton
+          label={<Trans>Already have an account?</Trans>}
+          primary={false}
+          key="already-have-account"
+          onClick={onGoToLogin}
+          disabled={createAccountInProgress}
+        />,
+      ]}
+      cannotBeDismissed={createAccountInProgress}
+      onApply={createAccount}
+      onRequestClose={() => {
+        if (!createAccountInProgress) onClose();
+      }}
+      maxWidth="sm"
+      open
+    >
+      <ColumnStackLayout noMargin>
+        <BackgroundText>
+          <MarkdownText
+            translatableSource={t`By creating an account and using GDevelop, you agree to the [Terms and Conditions](https://gdevelop.io/page/terms-and-conditions). Having an account allows you to export your game on Android or as a Desktop app and it unlocks other services for your project!`}
           />
-        </ColumnStackLayout>
-      </Dialog>
-    );
-  }
-}
+        </BackgroundText>
+        <UsernameField
+          value={username}
+          onChange={(e, value) => {
+            setUsername(value);
+            if (!value) {
+              // User can create an account without a username.
+              setUsernameAvailability(null);
```

Comment:
Isn't it handled by the UsernameField?

Reply:
yep, I had a previous version where it wasn't. I'll remove it 👍